enhancements/builders.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Removing Bare-metal Requirement For Builds
+authors:
+  - "@bcaton"
+reviewers:
+  - "@kleesc"
+  - "@bcaton"
+  - "@harishsurf"
+approvers:
+  - "@kleesc"
+  - "@bcaton"
+  - "@harishsurf"
+creation-date: 2021-12-03
+last-updated: 2021-12-03
+status: implementable
+---
+
+# remove-bare-metal-build-constraint
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+> 1. Will we encounter issues down the line running builds in unpriviliged containers with Podman?
+> 1. Does running builds in unprivileged containers give us the build isolation we need?
+
+## Summary
+
+Currently Quay builds run Podman commands in virtual machines launched via Pods. To run builds on virtual platforms this would require enabling nested virtualization, which is not possible in RHEL or OCP. This places the requirement of builds having to run on bare-metal clusters, which are a precious resource. This proposol is to remove the bare-metal constraint required to run builds.
+
+### Goals
+
+- Builds are able to run on virtualized platforms
+- Have backwards-compability to run previous build configurations
+
+## Design Details
+
+This solution proposes removing the VM layer to permit builds on virtualized platforms. An unprivileged Podman container will be used to run the image build.
+
+This diagram describes the current workflow:
+
+![image](https://user-images.githubusercontent.com/22058392/146998028-60e79958-5ef8-49ea-af07-7c9aaeef938a.png)
+
+The following diagram describes the proposed workflow:
```

Comment:
What are the benefits of doing this in house vs leveraging something like Tekton? Faster and easier to deliver?

Reply:
Removing the VM constraint requires straight forward code changes. Integrating Tekton would require overhauling the current build process. Tekton solves the issue of us handling the current build workflow, but is not necessary for removing the build constraint. We're currently seeing these as two different solutions that are solving two separate problems.